[package] run_depend libusb-1.0 -> libusb-1.0-dev

moriarty opened the pull request:
Change run_depend to libusb-1.0-dev

CMake Error: Project 'sick_tim' specifies '/usr/include/libusb-1.0' as an include dir, which is not found.

It looks like libusb headers are included in the headers exported by this project, so the **-dev** 
version should be a `<run_depend>` (or `<build_export_depend>` after [REP-140](http://www.ros.org/reps/rep-0140.html)) 

This error occurred on the ROS build farm, oddly on our doc job not the other jobs.
http://build.ros.org/job/Mdoc__fetch_robots__ubuntu_bionic_amd64/2/console

Full error:
```
03:23:52 -- Using these message generators: gencpp;geneus;genlisp;gennodejs;genpy
03:23:52 CMake Error at /opt/ros/melodic/share/sick_tim/cmake/sick_timConfig.cmake:113 (message):
03:23:52   Project 'sick_tim' specifies '/usr/include/libusb-1.0' as an include dir,
03:23:52   which is not found.  It does neither exist as an absolute directory nor in
03:23:52   '/opt/ros/melodic//usr/include/libusb-1.0'.  Check the website
03:23:52   'http://wiki.ros.org/sick_tim' for information and consider reporting the
03:23:52   problem.
```

moriarty committed:
[package] run_depend libusb-1.0 -> libusb-1.0-dev 

Change run_depend to libusb-1.0-dev

CMake Error: Project 'sick_tim' specifies '/usr/include/libusb-1.0' as an include dir, which is not found.

jspricke commented:
run_depend sounds wrong to me, build_export_depend sounds right. Any reason why you didn't use it?

mintar commented:
I agree, this should be a build_export_depend.

mintar commented:
I'm quite busy this week because of the IROS deadline on Friday, but next week I'll add a build_export_depend and re-release sick_tim.

moriarty commented:
I can submit a PR with `build_export_depend` but the reason I didn't is because this package isn't yet using the REP-140, so I'd need to go through and change the package.xml to format 2

I'm flying to Koeln for Karneval tomorrow ;) but I can try to submit a PR in the meantime

moriarty committed:
REP-140 package.xml format="2"

See: http://www.ros.org/reps/rep-0140.html

Switch from REP-127 to REP-140 for <build_export_depend>.

moriarty commented:
I've pushed an updated commit and the pre-release test is in progress.

```
mkdir -p /tmp/prerelease_job
cd /tmp/prerelease_job
generate_prerelease_script.py \
  https://raw.githubusercontent.com/ros-infrastructure/ros_buildfarm_config/production/index.yaml \
  melodic default ubuntu bionic amd64 \
  --custom-repo \
    sick_tim__custom-3:git:https://github.com/moriarty/sick_tim.git:patch-1 \
  --level 1 \
  --output-dir ./
```

moriarty committed:
FIX: build_export_depend and build_depend

The Jenkins PR job failed.

Both build_depend and build_export_depend are needed.

moriarty commented:
@jspricke & @mintar feel free to squash if/when you merge.

I believe this change should be compatible with indigo/lunar/kinetic and melodic. So you should be able to fast-forward the other branches and not need to create a fork point in the git history, and bloom <0.0.15> to all distros

I've re-started the prerelease script with indigo (passed) and kinetic (still running)... I skipped Lunar.

jspricke commented on package.xml:
```diff
@@ -17,19 +17,15 @@
+  <depend>roscpp</depend>
+  <depend>sensor_msgs</depend>
+
+  <exec_depend>libusb-1.0-dev</exec_depend>
```
I would propose to switch this to libusb-1.0

moriarty replied:
of course, that was an oversight. Done.

jspricke reviewed (commented):
thanks, fine with me except the comment

moriarty commented on package.xml:
```diff
@@ -18,18 +18,17 @@
+  <depend>roscpp</depend>
+  <depend>sensor_msgs</depend>
+
+  <exec_depend>libusb-1.0-dev</exec_depend>
```
```suggestion
  <exec_depend>libusb-1.0</exec_depend>
```

moriarty committed:
exec_depend libusb non-dev (review feedback)

moriarty commented:
Anything holding up the merge?

This causes our docs job to fail and send me an email every night.

mintar commented:
Thanks for the PR! Merged. Will do the release on Monday, you'll probably continue to get those mails until after the release. Enjoy Karneval!

mintar commented:
I've done the release.
